creusot/src/translation/pearlite.rs

```diff
@@ -202,7 +202,7 @@ pub(crate) fn pearlite(tcx: TyCtxt, id: LocalDefId) -> CreusotResult<Term> {
         return Err(Error::new(tcx.def_span(id), "type checking failed"));
     };
 
-    visit::walk_expr(&mut PurityVisitor { tcx, thir: &thir }, &thir[expr]);
+    visit::walk_expr(&mut PurityVisitor { tcx, thir: &thir, in_pure_ctx: true }, &thir[expr]);
```

Comment:
Maybe we can move this check out of here and just do it at the same time for every function?